lib/test/Crunch_Test_NonDet.thy

```diff
@@ -26,13 +26,14 @@ definition
     crunch_foo1 13
   od"
 
+crunch_ignore (valid, empty_fail, no_fail) (add: bind)
```

Comment:
wonderful demo of this working

Reply:
Oh yeah, I was going to ask for people's opinions on the syntax.

Is everyone happy for both parts of that invocation to use parentheses? While I'm here we could easily change the first to use a different pair of brackets if it might make it look nicer.

Reply:
I'm fine to leave it as is. If we switched it to `[ ]`, would that make it any clearer what it does? I guess it would distinguish part 1 (props) from part 2 (add/del instructions). Do you have any feelings on this?

Reply:
I think I agree. It would be nice to distinguish the two parts, but all the options I've thought of either have other problems or don't seem to help that much.

I guess the 'nicest' way might be `crunch_ignore ‹valid, empty_fail, no_fail› (add: bind)`, but I'm not sure how standard that is and it barely feels worth changing.

Reply:
Agreed. Can check in with Gerwin tomorrow to see if he has any thoughts, but looks like current syntax will be perfectly fine.

Reply:
I don't think either of them (`[ .. ]` and `‹ .. ›`) are good, because both are associated with different existing concepts in Isabelle already (`[ .. ]` for attributes and `‹ .. ›` for changing syntax context, e.g. to antiquotation or inner syntax).

We could do what we do for `crunches`, though: `crunch_ignore valid, empty_fail, no_fail (...)`, or even `crunch_ignore ops for valid, empty_fail, no_fail` although that latter would be a larger break with current `crunch_ignore` syntax.

Reply:
I prefer leaving it as is over those options. I can't find a nice way for both `crunches` and `crunch_ignore` to have similar syntaxes and still be nicely readable. Here's an example of a `crunches` command.
```
crunches readVCPUHardwareReg
  for (no_fail) no_fail[intro!, wp, simp]
  and (empty_fail) empty_fail[intro!, wp, simp]
  and (no_irq) no_irq[intro!, wp, simp]
  ```

The only option that would be consistent is something like your `crunch_ignore ops for valid, empty_fail, no_fail` suggestion, but then that doesn't have a nice way for removing things from the `crunch_ignore` set.